extensions/lifecycle-livedata-preference/src/main/java/com/twocoders/extensions/lifecycle/livedata/preference/SharedPreferencesLiveData.kt

```diff
@@ -0,0 +1,51 @@
+package com.twocoders.extensions.lifecycle.livedata.preference
+
+import android.content.SharedPreferences
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+@Suppress("MemberVisibilityCanBePrivate")
+abstract class SharedPreferencesLiveData<T : Any>(
+    protected val prefs: SharedPreferences,
+    protected val prefKey: String,
+    protected val defValue: T
+) : MutableLiveData<T>() {
+
+    abstract fun getValueFromPrefs(): T
+    abstract fun setValueToPrefs(newValue: T)
+
+    private val preferenceChangeListener = SharedPreferences.OnSharedPreferenceChangeListener { _, key ->
+        if (key == this.prefKey) {
+            CoroutineScope(Dispatchers.Main).launch {
```

Comment:
nechces namiesto zmeny scope pouzit `postValue`?

Reply:
`postValue` jsme tu myslím zkoušeli, s klasickými SharedPreferences to fungovalo, ale problém byl s EncryptedSharedPreferences. Mají ten interface jinak implementovaný a tak se to s `postValue` myslím zacyklilo :/

Reply:
mozno prave pre to `if (value != newValue) setValueToPrefs(newValue)`. Ak by sa volal super, **mozno** by to necyklilo

Reply:
však si to prosím nejprve vyzkoušej u sebe s oběma implementacemi než začneš navrhovat úpravy z hlavy ;) už je o nějaký čas přesně si to nepamatuji, jen si vzpomínám, že tou "čistou cestou" to nefungovalo dobře s těmi EncryptedSharedPreferences

Reply:
no ok, len ja kym sa k tomu dostanem a este aj urobim example s `EncryptedSharedPreferences` tak to nebude hned

Reply:
problém byl ale Míro s `postValue` nikoli se `setValue`, ale na to test nevidím.

Reply:
tak jsi měl pravdu, funguje to i s `EncryptedSharedPreferences`, zkoušel jsem to i když se na tom observuje (v tu chvilku je totiž aktivní i ten preferenceChangeListener) v čase volání `postValue` a funguje to, nezacyklí se to 👍

Reply:
Ok super. No ja som myslel ze ide o ten interny `post`, ktory sa zavola pri zmene preferences a na to test bol. Samotne volanie `post` zvonku nie, to mas pravdu, to som nepocital, ze na to si myslel. Anyway uvidime, keby bol nejaky problem tak doplnime dalsie testy. 👍 😉